plugins/modules/credential.py

```diff
@@ -201,6 +220,23 @@ def main() -> None:
         else (controller.get_item_name(credential) if credential else name)
     )
 
+    # we attempt to copy only when copy_from is passed
+    if copy_from:
+        try:
+            result = controller.copy_if_needed(
+                name,
+                copy_from,
+                endpoint=f"eda-credentials/{credential['id']}/copy",
+                item_type="credential",
```

Comment:
This is only a suggestion. Since your passing item_type to copy_if_needed(), why don't you build the endpoint inside the function based on the item_type?

For example: if item_type == "credential" -> endpoint = f"eda-credentials/{id}/copy" elif item_type == "activation" -> endpoint = f"activations/{activation['id']}/copy"

You will have copy_if_needed(name, copy_from, id, item_type).

Reply:
I think moving the endpoint into the copy_if_needed function would be fine for now, but thinking for future scenarios it might get difficult to scale it, for example, if we add copy feature for other resources then we will end up with a big block of "if elif" code. Thoughts?

Reply:
I agree with Doston. In this case, we would end up having a big if elif block.

What we can do is to pass the endpoint as plain `endpoint="eda-credentials"` to `controller.copy_if_needed`, and, inside the function, retrieve the resource id with something like `id = controller.get_exactly_one` and finally format the endpoint like `endpoint = '{0}/{1}/{2}'.format(endpoint, id, "copy")` before sending the request `response = self.post_endpoint(endpoint, data={"name": name})`.

That way we keep the `item_type` behavior for informational purposes and guarantee it will work for other resources.

WDYT?

Reply:
get_exactly_one()  is already called here https://github.com/ansible/event-driven-ansible/pull/399/files#diff-e8e6c3361f7f035f239fecccb755edfa684f11c90374ecf6abac2cb11b61d408R199, so it will be called two times then.
I am fine with the implementation as it is and we can think about improvement later. I don't want to block the merging of this PR.